components/match2/commons/match_group_input.lua

```diff
@@ -315,7 +315,7 @@ Merges an opponent struct into a match2 opponent record.
 ]]
 function MatchGroupInput.mergeRecordWithOpponent(record, opponent)
 	if opponent.type == Opponent.team then
-		record.template = record.template or opponent.template
```

Comment:
Add a comment explaining why this order is important so it doesn't happen again

Reply:
Comment added to the function comment, do you think it's sufficient?